build-system/tasks/runtime-test/runtime-test-base.js

```diff
@@ -173,10 +179,51 @@ function updateReporters(config) {
   }
 }
 
+/**
+ * Prints a dot for every babel transform, with wrapping if needed.
+ */
+function printBabelDot() {
+  process.stdout.write('.');
+  if (++wrapCounter >= dotWrappingWidth) {
+    wrapCounter = 0;
+    process.stdout.write('\n');
+  }
+}
+
+/**
+ * Constructs the configuration object used by esbuild for test transforms.
+ * @return {!Object}
+ */
+function getEsbuildConfig() {
+  const importPathPlugin = {
+    name: 'import-path',
+    setup(build) {
+      build.onResolve({filter: /^[\w-]+$/}, (file) => {
+        if (file.path === 'stream') {
+          return {path: require.resolve('stream-browserify')};
+        }
+      });
+    },
+  };
+  const babelPlugin = getEsbuildBabelPlugin(
+    'test',
+    /* enableCache */ true,
+    printBabelDot
+  );
+  return {
+    inject: ['./test/_init_tests.js'],
```

Comment:
@jridgewell Do tests actually run on your machine with these changes? I'm seeing all transforms complete successfully, after which Karma doesn't do anything, and Chrome devtools is unresponsive. Looks like we need to figure out what is keeping Karma busy.

Reply:
So, this it wasn't actually Karma but esbuild + karma-esbuild. Every single file that matches it treated as a endpoint file, and so we compile all JS code for that binary. So we ended up with several hundred compilations of the entire AMP codebase, one for each test file, and tried feeding that to Karma to run. On my  machine, it was 20GB of RAM before it failed to allocate more memory.

Reply:
Yup, @jridgewell  is right on the money here. Just doing a full bundle for each matched file was easier to do. I mainly made `karma-esbuild` to speed up the testing compilation time over at [Preact](https://github.com/preactjs/preact) by switching away from webpack. For this purpose it was fast enough to just do a full compile as it's `<1s` for file changes. Scale that up to the amount of tests and files you have here obviously requires a more granular approach. Chatted with Justin on Slack the other day and I happy merge any improvements you have on that front 👍